python/lsst/daf/butler/_quantum_backed.py

```diff
@@ -352,12 +352,15 @@ def _initialize(
                 datasets=_DatasetRecordStorageManagerDatastoreConstructionMimic,  # type: ignore
                 universe=dimensions,
             )
-        # TODO: We need to inform `Datastore` here that it needs to support
-        # predictive reads; right now that's a configuration option, but after
-        # execution butler is retired it could just be a kwarg we pass here.
-        # For now just force this option as we cannot work without it.
-        butler_config["datastore", "trust_get_request"] = True
         datastore = Datastore.fromConfig(butler_config, bridge_manager, butler_root)
+
+        # TODO: We need to inform `Datastore` here that it needs to support
+        # predictive reads; This only really works for file datastore but
+        # we need to try everything in case there is a chained datastore.
+        for this_datastore in getattr(datastore, "datastores", [datastore]):
+            if hasattr(this_datastore, "trustGetRequest"):
+                this_datastore.trustGetRequest = True
```

Comment:
So, if by some chance non-chained datastore adds `datastores` attribute, we won't try to set its `trustGetRequest`? And it will also crash if `datastore.datastores` is not an Iterable?

Reply:
If `datastores` is not an iterable we are in trouble indeed. I was pondering adding `datastores` to `Datastore` and making it return `[self]` by default. Then it would be clear that it has a special purpose that only something like `ChainedDatastore` can override.

Reply:
Maybe just add "private" method `_set_trust_mode()` to the base class?

Reply:
I will do that. I wasn't sure I wanted to make it something that top-level given we can't think of anything outside FileDatastore caring.